app/services/nomis/client.rb

```diff
@@ -87,15 +91,15 @@ def http_method_expects(method)
       else
         # TODO: Change 400 to 422 when api is updated.
         # Currently it returns 400 for validation errors.
-        [200, 400, 409]
+        [200, 400, 404, 409]
```

Comment:
That's the error code we get when the visit has already been cancelled? Although the API documentation specifies it should return a 400. I'll file a bug in the NOMIS API board.

Have a look at this cassette:
spec/fixtures/vcr_cassettes/cancelled_visit.yml

Reply:
I have looked at it, I don't think the test setup is/was correct. The error message is:

`'{"error":{"message":"Visit not found"}}'`

Reply:
I think you're confused between an *invalid cancellation code* and a *visit already cancelled*

Reply:
so until the NOMIS API does not return a 400 for a visit already cancelled we can't release this feature

Reply:
Yes, correct. We can't expect a 404 code when the visit is already cancelled.

Reply:
Looked at the cassette, and is not very clear the setup.

Is the assumption that the same api endpoint was called before? 

Since there is only one call and the API returns a 404 with a specific error message for the visit id it makes it look like the visit id is invalid.

For me the intent of the cassette would be clearer if the cassette had 2 requests, the first showing the visit getting cancelled and then the second request failing. What do you think?